misc/package/Data Files/MWSE/core/lib/logger/init.lua

```diff
@@ -0,0 +1,866 @@
+---@diagnostic disable: duplicate-set-field
+--[[ # Logger 
+
+Based on the `mwseLogger` made by Merlord.
+
+Current version made by herbert100.
+
+Many suggestions and feature ideas given by C3pa.
+
+Credit to Greatness7 as well, for the ideas of adding a `format` parameter and allowing functions to be passed to log messages.
+]]
+
+local socket = require("socket")
+
+-- The time that the game was launched.
+---@type number
+local LAUNCH_TIME = socket.gettime()
+
+
+local fmt = string.format
+
+
+
+
+local LOG_LEVEL = {
+	NONE  = 0,
+	ERROR = 1,
+	WARN  = 2,
+	INFO  = 3,
+	DEBUG = 4,
+	TRACE = 5
+}
```

Comment:
Sorry about not mentioning this earlier, but as [G7 remarked](https://github.com/MWSE/MWSE/pull/462#issuecomment-1868127985) at the time this PR was opened, please rename this enumeration to camel case:
```lua
local logLevel = {
        none = 0,
        error = 1,
        -- etc.
}
```